plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java

```diff
@@ -106,130 +107,121 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<LoadMo
-                        eligibleNodes.add(nodeMapping.get(nodeId));
-                        nodeIds.add(nodeId);
-                    }
+        List<DiscoveryNode> eligibleNodes = new ArrayList<>();
+        List<String> nodeIds = new ArrayList<>();
+        if (targetNodeIds != null && targetNodeIds.length > 0) {
+            for (String nodeId : targetNodeIds) {
+                if (allEligibleNodeIds.contains(nodeId)) {
+                    eligibleNodes.add(nodeMapping.get(nodeId));
+                    nodeIds.add(nodeId);
                 }
-            } else {
-                nodeIds.addAll(allEligibleNodeIds);
-                for (DiscoveryNode node : allEligibleNodes) {
-                    eligibleNodes.add(node);
-                }
-            }
-            if (nodeIds.size() == 0) {
-                listener.onFailure(new MLResourceNotFoundException("no eligible node found"));
-                return;
             }
+        } else {
+            nodeIds.addAll(allEligibleNodeIds);
+            eligibleNodes.addAll(Arrays.asList(allEligibleNodes));
+        }
+        if (nodeIds.size() == 0) {
+            listener.onFailure(new MLResourceNotFoundException("no eligible node found"));
+            return;
+        }
+
+        String workerNodes = String.join(",", nodeIds);
+        log.warn("Will load model on these nodes: {}", workerNodes);
+        String localNodeId = clusterService.localNode().getId();
 
-            String workerNodes = String.join(",", nodeIds);
-            log.warn("Will load model on these nodes: {}", workerNodes);
-            String localNodeId = clusterService.localNode().getId();
-
-            String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
-            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-                mlModelManager.getModel(modelId, null, excludes, ActionListener.wrap(mlModel -> {
-                    FunctionName algorithm = mlModel.getAlgorithm();
-                    // TODO: Track load failure
-                    // mlStats.createCounterStatIfAbsent(algorithm, ActionName.LOAD, MLActionLevelStat.ML_ACTION_REQUEST_COUNT).increment();
-                    MLTask mlTask = MLTask
-                        .builder()
-                        .async(true)
-                        .modelId(modelId)
-                        .taskType(MLTaskType.LOAD_MODEL)
-                        .functionName(algorithm)
-                        .createTime(Instant.now())
-                        .lastUpdateTime(Instant.now())
-                        .state(MLTaskState.CREATED)
-                        .workerNode(workerNodes)
-                        .build();
-                    mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
-                        String taskId = response.getId();
-                        mlTask.setTaskId(taskId);
-
-                        try {
```

Comment:
I think we should keep this try block, this was to catch exception of different parts, will print different exception message and also remove task `mlTaskManager.remove(taskId);`

Reply:
Added this try catch back.